internal/twirptest/service_test.go

```diff
@@ -1527,3 +1527,123 @@ func TestPanicFlushing(t *testing.T) {
 		t.Errorf("twirp client err has unexpected message %q, want %q", twerr.Msg(), "Internal service panic")
 	}
 }
+
+type errReader int
```

Comment:
Optional: It looks like the value of this never gets used (but looks in some places like it _might_). I'd make this a `struct{}` so it's clear that it holds no useful information.

And an extra aside: I'd suggest also https://golang.org/pkg/testing/iotest/#ErrReader , but I'm surprised to see that it's new in Go 1.16, so is still too new to use in Twirp (we need to support at least the previous Go version, and maybe the one before that too).